### PR TITLE
Support for Yandex Object Storage (#144)

### LINKED ISSUE #144: Support for Yandex Object Storage
I tried using minio-hs with [Yandex Object Storage](https://cloud.yandex.com/docs/storage/)
and got an error: 
```
uncaught exception: MinioErr
MErrHTTP (HttpExceptionRequest Request {
  host                 = "storage.yandexcloud.net"
  port                 = 443
  secure               = True
  requestHeaders       = [("Host","storage.yandexcloud.net"),("Authorization","<REDACTED>"),("X-Amz-Date","20200518T171120Z"),("x-amz-content-sha256","UNSIGNED-PAYLOAD")]
  path                 = "/bucket-test/piPeo"
  queryString          = "uploads"
  method               = "POST"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
 (StatusCodeException (Response {responseStatus = Status {statusCode = 403, statusMessage = "Forbidden"}, responseVersion = HTTP/1.1, responseHeaders = [("Server","nginx"),("Date","Mon, 18 May 2020 17:11:21 GMT"),("Content-Type","application/xml; charset=UTF-8"),("Transfer-Encoding","chunked"),("Connection","keep-alive"),("Keep-Alive","timeout=60"),("X-Amz-Request-Id","5ca10399c0567052"),("Content-Encoding","gzip")], responseBody = (), responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}) "Response {responseStatus = Status {statusCode = 403, statusMessage = \"Forbidden\"}, responseVersion = HTTP/1.1, responseHeaders = [(\"Server\",\"nginx\"),(\"Date\",\"Mon, 18 May 2020 17:11:21 GMT\"),(\"Content-Type\",\"application/xml; charset=UTF-8\"),(\"Transfer-Encoding\",\"chunked\"),(\"Connection\",\"keep-alive\"),(\"Keep-Alive\",\"timeout=60\"),(\"X-Amz-Request-Id\",\"5ca10399c0567052\"),(\"Content-Encoding\",\"gzip\")], responseBody = \"<?xml version='1.0' encoding='UTF-8'?>\\n<Error><Code>AccessDenied</Code><Message>There were headers present in the request which were not signed</Message><Resource>/bucket-test/piPeo</Resource><RequestId>5ca10399c0567052</RequestId><HeadersNotSigned>x-amz-content-sha256</HeadersNotSigned></Error>\", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}"))
```
Key message here is:
> There were headers present in the request which were not signed
> <HeadersNotSigned> x-amz-content-sha256 </HeadersNotSigned>

I believe this can be fixed by adding the variable [`sha256Hdr`](https://github.com/minio/minio-hs/blob/97393762275408f67c2efeb5a00ce7874b918fe9/src/Network/Minio/Sign/V4.hs#L161) to the list [`headersToSign`](https://github.com/minio/minio-hs/blob/97393762275408f67c2efeb5a00ce7874b918fe9/src/Network/Minio/Sign/V4.hs#L127)

### PR DESCRIPTION
**Pull request by ivanovs-4**
Fixes #144

### CODE REVIEW

File: src/Network/Minio/Sign/V4.hs [hunk 1]
```diff
@@ -125,6 +127,7 @@ signV4 !sp !req =
                       then []
                       else [(\(x, y) -> (mk x, y)) datePair]
     headersToSign = getHeadersToSign computedHeaders
+                 <> [sha256Hdr]
```
Comment (donatello) [inline src/Network/Minio/Sign/V4.hs:130]: This should be the right fix:
```suggestion
                      else map (\(x, y) -> (mk x, y)) [datePair, sha256Hdr]
    headersToSign = getHeadersToSign computedHeaders
```

**Comment by donatello** [comment]
@ivanovs-4 Could you apply this change and verify that it fixes your problem?

**Comment by donatello** [comment]
Though the build fails, if it fixes your problem, we can take this in. We will take up fixing the build and do some cleanup and make a release.

**Comment by donatello** [comment]
This will pass CI after #146 is merged

**Comment by donatello** [comment]
@ivanovs-4 Please update this change so it is up to date with master - I would do this, but it looks like your settings don't allow me to do this.

**Comment by donatello** [comment]
Took this change in via #147 - thank you for the fix @ivanovs-4

**Comment by ivanovs-4** [comment]
@donatello,
it turns out that this change may violate compatibility with the original minio server with an error:
MErrService (ServiceErr "SignatureDoesNotMatch" "The request signature we calculated does not match the signature you provided. Check your key and signing method.")
or i may be wrong. I will check it twice.

**Comment by ivanovs-4** [comment]
Well. it works actually.
Thank you for the library @donatello

**Pull request closed without merging**